src/DefaultBuilder/src/WebHostEnvironment.cs

```diff
@@ -100,19 +93,81 @@ public void ResolveFileProviders(IConfiguration configuration)
                 WebRootFileProvider = new PhysicalFileProvider(WebRootPath);
             }
 
-            if (this.IsDevelopment())
-            {
-                StaticWebAssetsLoader.UseStaticWebAssets(this, configuration);
-            }
         }
 
         public string ApplicationName { get; set; }
         public string EnvironmentName { get; set; }
 
-        public IFileProvider ContentRootFileProvider { get; set; }
-        public string ContentRootPath { get; set; }
+        private IFileProvider _contentRootFileProvider = NullFileProvider;
+        public IFileProvider ContentRootFileProvider
+        {
+            get => _contentRootFileProvider;
+            set
+            {
+                if (_contentRootPath == default(string))
```

Comment:
Nit:

```suggestion
                if (_contentRootPath is null)
```

If we keep this check, we should add a test for it. I'm not sure we need it though. `_contentRootPath` is set by the ctor, and this doesn't catch all the ways the file provider and path could become out of sync anyway.

Reply:
Per our discussion, we're removing this check to account for edge cases that would not be accounted for. This means that it's up to the user to ensure that the provider and path are consistent but it's not likely the user will intentionally misset the provider here.